lib/src/utils/utils.dart

```diff
@@ -1,3 +1,4 @@
+/// parses unique matrix `userId` and return name.
```

Comment:
until now everything in avatar was kinda generic, didn't expect or assume matrix as the underlying concept ... This introduces a hard dependency internally without exposing that the input (on the widget) will not work as expected if the format doesn't match.

I'd suggest that rather than `null` this returns `userId` unchanged in non-matrix-id cases ... but more over I am not sure I agree this should just implicitly happen inside the widget ... Maybe instead we want to provide this as an external function and recommend using that for the input value ...

Reply:
I see, okay I'll move this out from internal package. I agree we shouldn't use this function implicitly but leave at the disposal of user if and how they want to handle that.